### PR TITLE
Updated tortoise-hg to current 2021 version

### PR DESCRIPTION
**Pull request by gracinet**
The previously listed versions were out of support for years.

Context: I am a Mercurial core contributor, currently working to provide Windows CI resources for the Mercurial and TortoiseHg projects, and been using Salt on Linux for years (with some minor contribs). 

Being able to manage these machines with Salt will be a drastic simplification to me so I'd like to thank you for making salt_winrepo_ng.

The new definition works for me on an amd64 (x64) VM with Windows Server 2019.

I didn't have the occasion to try it on x86, that's why I kept a general `else` Jinja clause instead of comparing to the actual architecture.

### CODE REVIEW

File: tortoise-hg.sls [hunk 1]
```diff
@@ -1,34 +1,16 @@
-# Source: http://tortoisehg.bitbucket.org/
+{% if grains['cpuarch'] == 'AMD64' %}
+{% set msi_arch = 'x64' %}
+{% else %}
+{% set msi_arch = 'x86' %}
+{% endif %}
+
 tortoise-hg:
```
Comment (twangboy) [inline tortoise-hg.sls:7]: We still need to keep the older versions around so that they can be uninstalled.

**Review by twangboy — CHANGES_REQUESTED** [review]

File: tortoise-hg.sls (hunk 1 above)
Comment (gracinet) [reply tortoise-hg.sls:7]: Didn't think of that, of course, makes sense. I pushed a correction.

Sadly the old 3.3.0 and 3.6.2 are not currently available from the official download site, so we have to hope that people that kept the MSIs.

I've asked the TortoiseHg maintainers.

**Comment by gracinet** [comment]
Also fixed a leftover `x64` in first commit

**Comment by TheBigBear** [comment]
In cases like, either because it had CVEs or old versions that are no longer available,  then just leave the uninstall related info and remove the install related info. 

( look at npp.sls and how it deals with versions prior to ver. 7.6.6 that are no longer publicly available. Or you could leave the installer URL in and do a skip open the test with the
"skip_urltest: True” line for those.

**Comment by gracinet** [comment]
@TheBigBear thanks for the review and the tips!

I've disabled the install for the old versions.

**Review by twangboy — APPROVED** [review]

**Comment by gracinet** [comment]
@twangboy great, many thanks!

**Pull request merged**